shepherd/storage/minio_storage.py

```diff
@@ -21,74 +24,89 @@
 """Minio folder delimiter."""
 
 
-def _botocore_code(error: BotocoreClientError) -> str:
-    """
-    Extract an error code from a botocore exception.
-
-    :param error: the error to extract
-    :return: the error code (sometimes a human-readable string, sometimes a number in string form)
-    """
-    return error.response.get("Error", {}).get("Code", "")
-
-
 class MinioStorage(Storage):
     """
     A remote storage adapter that uses the aiobotocore S3 client to access Minio.
     """
 
+    _NS = {
+        "s3": "http://s3.amazonaws.com/doc/2006-03-01/"
+    }
+
     def __init__(self, storage_config: StorageConfig):
         """
         Initialize the storage according to the configuration.
 
         :param storage_config: storage configuration
         """
 
-        session = aiobotocore.get_session(loop=asyncio.get_event_loop())
-        self._client = session.create_client('s3',
-                                             endpoint_url=storage_config.url,
-                                             use_ssl=storage_config.secure,
-                                             aws_access_key_id=storage_config.access_key,
-                                             aws_secret_access_key=storage_config.secret_key)
+        self._session = aiohttp.ClientSession()
+        self._config = storage_config
+
+    @staticmethod
+    def _create_headers(headers: Optional[LooseHeaders] = None) -> LooseHeaders:
+        """Add User-Agent header if not specified yet."""
+        if headers is None:
+            headers = {}
+
+        headers.setdefault("User-Agent", "Shepherd")
+
+        return headers
+
+    def _add_auth_header(self, method: str, url: str, headers: Optional[LooseHeaders] = None,
+                         content_sha256: Optional[str] = None) -> LooseHeaders:
+        if headers is None:
```

Comment:
This line is unnecessary if you modify the following one to
`headers = MinioStorage._create_headers(headers)`